<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QWidget" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1402</width>
    <height>847</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="GLWidget_2D" name="glWid2dim" native="true">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>10</y>
     <width>621</width>
     <height>451</height>
    </rect>
   </property>
   <widget class="Line" name="line_4">
    <property name="geometry">
     <rect>
      <x>-10</x>
      <y>0</y>
      <width>20</width>
      <height>451</height>
     </rect>
    </property>
    <property name="orientation">
     <enum>Qt::Vertical</enum>
    </property>
   </widget>
   <widget class="Line" name="line_6">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>-10</y>
      <width>431</width>
      <height>16</height>
     </rect>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
  </widget>
  <widget class="QGroupBox" name="CurveListBox">
   <property name="geometry">
    <rect>
     <x>520</x>
     <y>490</y>
     <width>151</width>
     <height>231</height>
    </rect>
   </property>
   <property name="title">
    <string>Curve</string>
   </property>
   <widget class="QWidget" name="verticalLayoutWidget">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>20</y>
      <width>121</width>
      <height>204</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout">
     <item>
      <widget class="QPushButton" name="curve_add">
       <property name="text">
        <string>Add Curve</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="delete_curve">
       <property name="text">
        <string>Delete Curve</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QComboBox" name="CurveTypeBox">
       <property name="currentText">
        <string>Line</string>
       </property>
       <item>
        <property name="text">
         <string>Line</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>B-spline</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>Arc</string>
        </property>
       </item>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="move_ctrl_pt">
       <property name="text">
        <string>Move Control Point</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="insert_ctrl_pt">
       <property name="text">
        <string>Insert Control Point</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="delete_ctrl_pt">
       <property name="text">
        <string>Delete Control Point</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QGroupBox" name="GradationBox">
   <property name="geometry">
    <rect>
     <x>350</x>
     <y>490</y>
     <width>161</width>
     <height>201</height>
    </rect>
   </property>
   <property name="title">
    <string>Gradation</string>
   </property>
   <widget class="QPushButton" name="AddPointsButton">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>30</y>
      <width>151</width>
      <height>24</height>
     </rect>
    </property>
    <property name="text">
     <string>add GradationPoints</string>
    </property>
   </widget>
   <widget class="QComboBox" name="CBox_InterpolationType">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>60</y>
      <width>151</width>
      <height>22</height>
     </rect>
    </property>
    <item>
     <property name="text">
      <string>linear</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>Spline</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>B-spline</string>
     </property>
    </item>
   </widget>
   <widget class="QWidget" name="horizontalLayoutWidget">
    <property name="geometry">
     <rect>
      <x>9</x>
      <y>100</y>
      <width>141</width>
      <height>51</height>
     </rect>
    </property>
    <layout class="QHBoxLayout" name="horizontalLayout_2">
     <item>
      <widget class="QLineEdit" name="CP_colortype"/>
     </item>
     <item>
      <widget class="QLineEdit" name="CP_colorval"/>
     </item>
    </layout>
   </widget>
   <widget class="QSlider" name="CP_colorSlider">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>170</y>
      <width>141</width>
      <height>21</height>
     </rect>
    </property>
    <property name="minimum">
     <number>-255</number>
    </property>
    <property name="maximum">
     <number>255</number>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
  </widget>
  <widget class="QGroupBox" name="OutlineBox">
   <property name="geometry">
    <rect>
     <x>20</x>
     <y>490</y>
     <width>161</width>
     <height>211</height>
    </rect>
   </property>
   <property name="title">
    <string>Outline</string>
   </property>
   <widget class="QWidget" name="verticalLayoutWidget_3">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>20</y>
      <width>121</width>
      <height>191</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout_3">
     <item>
      <widget class="QCheckBox" name="outline_rectangle">
       <property name="text">
        <string>Rectangle</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QCheckBox" name="outline_polyline">
       <property name="text">
        <string>PolyLine</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QCheckBox" name="ConnectVertices">
       <property name="text">
        <string>Connect Vertices</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QCheckBox" name="outline_polygon">
       <property name="text">
        <string>Polygon</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QSpinBox" name="Polygon_EdgeNum">
       <property name="minimum">
        <number>3</number>
       </property>
       <property name="maximum">
        <number>10</number>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QCheckBox" name="MoveOutLineButton">
       <property name="text">
        <string>Move Outline</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QCheckBox" name="EditVertexButton">
       <property name="text">
        <string>Edit Vertex</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QGroupBox" name="DivideSizeBox">
   <property name="geometry">
    <rect>
     <x>210</x>
     <y>560</y>
     <width>121</width>
     <height>131</height>
    </rect>
   </property>
   <property name="title">
    <string>Divde Size</string>
   </property>
   <widget class="QWidget" name="verticalLayoutWidget_4">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>20</y>
      <width>101</width>
      <height>71</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout_4">
     <item>
      <widget class="QSpinBox" name="DivSizeSpinBox">
       <property name="minimum">
        <number>2</number>
       </property>
       <property name="maximum">
        <number>300</number>
       </property>
       <property name="value">
        <number>2</number>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QSlider" name="DvidedSizeSlider">
       <property name="minimum">
        <number>2</number>
       </property>
       <property name="maximum">
        <number>300</number>
       </property>
       <property name="value">
        <number>2</number>
       </property>
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QGroupBox" name="OtherBox">
   <property name="geometry">
    <rect>
     <x>840</x>
     <y>490</y>
     <width>171</width>
     <height>241</height>
    </rect>
   </property>
   <property name="title">
    <string>Other</string>
   </property>
   <widget class="QWidget" name="verticalLayoutWidget_2">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>20</y>
      <width>151</width>
      <height>216</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout_2">
     <item>
      <widget class="QCheckBox" name="SelectButton">
       <property name="text">
        <string>Select</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QCheckBox" name="make_devsrf">
       <property name="text">
        <string>Apply</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QCheckBox" name="Reset">
       <property name="text">
        <string>Reset</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="DebugWindow">
       <property name="text">
        <string>Debug</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="SaveButton">
       <property name="text">
        <string>Save</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QDoubleSpinBox" name="LineWidthSpinBox">
       <property name="decimals">
        <number>1</number>
       </property>
       <property name="minimum">
        <double>0.100000000000000</double>
       </property>
       <property name="maximum">
        <double>10.000000000000000</double>
       </property>
       <property name="singleStep">
        <double>0.100000000000000</double>
       </property>
       <property name="value">
        <double>1.000000000000000</double>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QSlider" name="LineWidthSlider">
       <property name="minimum">
        <number>1</number>
       </property>
       <property name="maximum">
        <number>100</number>
       </property>
       <property name="singleStep">
        <number>1</number>
       </property>
       <property name="pageStep">
        <number>0</number>
       </property>
       <property name="value">
        <number>10</number>
       </property>
       <property name="sliderPosition">
        <number>10</number>
       </property>
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QCheckBox" name="gridCBox">
       <property name="text">
        <string>erase grid</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QGroupBox" name="LayerListWidget">
   <property name="geometry">
    <rect>
     <x>1140</x>
     <y>460</y>
     <width>141</width>
     <height>311</height>
    </rect>
   </property>
   <property name="title">
    <string>Layer</string>
   </property>
  </widget>
  <widget class="GLWidget_3D" name="glWid3dim" native="true">
   <property name="geometry">
    <rect>
     <x>640</x>
     <y>10</y>
     <width>631</width>
     <height>441</height>
    </rect>
   </property>
  </widget>
  <widget class="Line" name="line_5">
   <property name="geometry">
    <rect>
     <x>-20</x>
     <y>400</y>
     <width>20</width>
     <height>451</height>
    </rect>
   </property>
   <property name="orientation">
    <enum>Qt::Vertical</enum>
   </property>
  </widget>
  <widget class="QGroupBox" name="GeometoryConstraitBox">
   <property name="geometry">
    <rect>
     <x>200</x>
     <y>490</y>
     <width>131</width>
     <height>61</height>
    </rect>
   </property>
   <property name="title">
    <string>Geometric Constraint</string>
   </property>
   <widget class="QPushButton" name="symmetryButton">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>30</y>
      <width>111</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>Symmetry</string>
    </property>
   </widget>
  </widget>
  <widget class="QGroupBox" name="FoldLineBox">
   <property name="geometry">
    <rect>
     <x>680</x>
     <y>490</y>
     <width>131</width>
     <height>151</height>
    </rect>
   </property>
   <property name="title">
    <string>Fold Line</string>
   </property>
   <widget class="QWidget" name="verticalLayoutWidget_5">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>30</y>
      <width>111</width>
      <height>138</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout_5">
     <item>
      <widget class="QPushButton" name="addFL_line">
       <property name="text">
        <string>Add Line</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="addFL_arc">
       <property name="text">
        <string>Arc</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="addFL_bezier">
       <property name="text">
        <string>Add Bezier</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="addFL_test">
       <property name="text">
        <string>Test Single Ruling</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="color_FL">
       <property name="text">
        <string>Color</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QSpinBox" name="planeBeta">
   <property name="geometry">
    <rect>
     <x>610</x>
     <y>460</y>
     <width>42</width>
     <height>22</height>
    </rect>
   </property>
   <property name="prefix">
    <string/>
   </property>
   <property name="maximum">
    <number>359</number>
   </property>
   <property name="value">
    <number>179</number>
   </property>
  </widget>
  <widget class="QLabel" name="label">
   <property name="geometry">
    <rect>
     <x>570</x>
     <y>460</y>
     <width>37</width>
     <height>12</height>
    </rect>
   </property>
   <property name="text">
    <string>beta</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_2">
   <property name="geometry">
    <rect>
     <x>470</x>
     <y>460</y>
     <width>31</width>
     <height>16</height>
    </rect>
   </property>
   <property name="text">
    <string>alpha</string>
   </property>
  </widget>
  <widget class="QSpinBox" name="planeBeta_2">
   <property name="geometry">
    <rect>
     <x>730</x>
     <y>460</y>
     <width>42</width>
     <height>22</height>
    </rect>
   </property>
   <property name="prefix">
    <string/>
   </property>
   <property name="maximum">
    <number>359</number>
   </property>
   <property name="value">
    <number>179</number>
   </property>
  </widget>
  <widget class="QLabel" name="label_4">
   <property name="geometry">
    <rect>
     <x>690</x>
     <y>460</y>
     <width>37</width>
     <height>12</height>
    </rect>
   </property>
   <property name="text">
    <string>beta2</string>
   </property>
  </widget>
  <widget class="QDoubleSpinBox" name="angleA">
   <property name="geometry">
    <rect>
     <x>500</x>
     <y>460</y>
     <width>62</width>
     <height>22</height>
    </rect>
   </property>
   <property name="maximum">
    <double>359.990000000000009</double>
   </property>
   <property name="singleStep">
    <double>0.010000000000000</double>
   </property>
  </widget>
 </widget>
 <customwidgets>
  <customwidget>
   <class>GLWidget_3D</class>
   <extends>QWidget</extends>
   <header>glwidget_3d.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>GLWidget_2D</class>
   <extends>QWidget</extends>
   <header>glwidget_2d.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
